casanovo/casanovo.py

```diff
@@ -130,13 +138,26 @@ def main(
 
     # Log the active configuration.
     logger.info("Casanovo version %s", str(__version__))
+    logger.debug("mode = %s", mode)
+    logger.debug("model = %s", model)
+    logger.debug("peak_dir = %s", peak_dir)
+    logger.debug("peak_dir_val = %s", peak_dir_val)
+    logger.debug("config = %s", config_fn)
+    logger.debug("output = %s", output)
     for key, value in config.items():
         logger.debug("%s = %s", str(key), str(value))
 
     # Run Casanovo in the specified mode.
     if mode == "denovo":
         logger.info("Predict peptide sequences with Casanovo.")
-        model_runner.predict(peak_dir, model, output, config)
+        _write_mztab_header(
+            f"{output}.mztab",
+            peak_dir,
+            config,
+            model=model,
+            config_filename=config_fn,
+        )
```

Comment:
**question:** Do you think we should clean up the file in the case that the predict step fails?

Reply:
Good suggestion to avoid confusion with incomplete files. I surrounded the predicted with a `try ... except` where the `expect` doesn't catch a specific error, which can be frowned upon. I think in this case it's probably warranted to have such a general `except` clause though. What do you think?

Reply:
Solved by refactoring the mzTab export functionality.